java-checks-test-sources/src/main/java/symbolicexecution/checks/DivisionByZeroCheckPrimitives.java

```diff
@@ -0,0 +1,76 @@
+package symbolicexecution.checks;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+@SuppressWarnings(/* indentation check */ "java:S1120")
+public abstract class DivisionByZeroCheckPrimitives {
+  private static final double DOUBLE_ZERO = 0.0;
+  private static final double DOUBLE_NOT_ZERO = 0.3048;
+  private static final Double DOUBLE_WRAPPER_ZERO = 0.0;
+  private static final Double DOUBLE_WRAPPER_NOT_ZERO = 0.3048;
+
+  private static final float FLOAT_ZERO = 0.0f;
+  private static final float FLOAT_NOT_ZERO = 0.3048f;
+  private static final Float FLOAT_WRAPPER_ZERO = 0.0f;
+  private static final Float FLOAT_WRAPPER_NOT_ZERO = 0.3048f;
+
+  private static final int INT_ZERO = 0;
+  private static final int INT_NOT_ZERO = 1;
+  private static final Integer INT_WRAPPER_ZERO = 0;
+  private static final Integer INT_WRAPPER_NOT_ZERO = 1;
+
+  private static final long LONG_ZERO = 0L;
+  private static final long LONG_NOT_ZERO = 1L;
+  private static final Long LONG_WRAPPER_ZERO = 0L;
+  private static final Long LONG_WRAPPER_NOT_ZERO = 1L;
+
+  private static final byte BYTE_ZERO = 0;
+  private static final byte BYTE_NOT_ZERO = 1;
+  private static final Byte BYTE_WRAPPER_ZERO = 0;
+  private static final Byte BYTE_WRAPPER_NOT_ZERO = 1;
+
+  private static final short SHORT_ZERO = 0;
+  private static final short SHORT_NOT_ZERO = 1;
+  private static final Short SHORT_WRAPPER_ZERO = 0;
+  private static final Short SHORT_WRAPPER_NOT_ZERO = 1;
+
+  private static final char CHAR_ZERO = 0;
+  private static final char CHAR_NOT_ZERO = 'a';
+
+  public int notZero(byte b)      { return b / BYTE_NOT_ZERO; }
+  public int notZero(short s)     { return s / SHORT_NOT_ZERO; }
+  public int notZero(int i)       { return i / INT_NOT_ZERO; }
+  public long notZero(long l)     { return l / LONG_NOT_ZERO; }
+  public double notZero(double d) { return d / DOUBLE_NOT_ZERO; }
+  public float notZero(float f)   { return f / FLOAT_NOT_ZERO; }
+  public int notZero(char c)      { return c / CHAR_NOT_ZERO; }
+  public BigInteger notZero(BigInteger bi) { return bi.divide(BigInteger.ONE); }
+  public BigDecimal notZero(BigDecimal bd) { return bd.divide(BigDecimal.ONE); }
```

Comment:
Below we are also testing both the `BigInteger.ZERO` and `BigInteger.valueOf(LONG_ZERO)` divisor cases − what do you think of including a test case along the lines of:
```
  public BigDecimal notZero2(BigDecimal bd) { return bd.divide(BigDecimal.valueOf(0.3048f)); }
```

Reply:
fair point!